modules/arkit/arkit_interface.mm

```diff
@@ -0,0 +1,711 @@
+		Transform transform;
+
+		matrix_float4x4 m44 = result.worldTransform;
+		transform.basis.elements[0].x = m44.columns[0][0];
+		transform.basis.elements[1].x = m44.columns[0][1];
+		transform.basis.elements[2].x = m44.columns[0][2];
+		transform.basis.elements[0].y = m44.columns[1][0];
+		transform.basis.elements[1].y = m44.columns[1][1];
+		transform.basis.elements[2].y = m44.columns[1][2];
+		transform.basis.elements[0].z = m44.columns[2][0];
+		transform.basis.elements[1].z = m44.columns[2][1];
+		transform.basis.elements[2].z = m44.columns[2][2];
+		transform.origin.x = m44.columns[3][0];
+		transform.origin.y = m44.columns[3][1];
+		transform.origin.z = m44.columns[3][2];
+
+		/* important, NOT scaled to world_scale !! */
+		arr.push_back(transform);
+	}
+
+	return arr;
+}
+
+void ARKitInterface::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("_notification", "what"), &ARKitInterface::_notification);
+
+	ClassDB::bind_method(D_METHOD("set_light_estimation_is_enabled", "enable"), &ARKitInterface::set_light_estimation_is_enabled);
+	ClassDB::bind_method(D_METHOD("get_light_estimation_is_enabled"), &ARKitInterface::get_light_estimation_is_enabled);
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "light_estimation"), "set_light_estimation_is_enabled", "get_light_estimation_is_enabled");
+
+	ClassDB::bind_method(D_METHOD("get_ambient_intensity"), &ARKitInterface::get_ambient_intensity);
+	ClassDB::bind_method(D_METHOD("get_ambient_color_temperature"), &ARKitInterface::get_ambient_color_temperature);
+
+	ClassDB::bind_method(D_METHOD("raycast", "screen_coord"), &ARKitInterface::raycast);
+}
+
+bool ARKitInterface::is_stereo() {
+	// this is a mono device...
+	return false;
+}
+
+bool ARKitInterface::is_initialized() const {
+	return (initialized);
+}
+
+bool ARKitInterface::initialize() {
+	ARVRServer *arvr_server = ARVRServer::get_singleton();
+	ERR_FAIL_NULL_V(arvr_server, false);
+
+	if (!initialized) {
+		print_line("initializing ARKit");
+
+		// create our ar session and delegate
+		ar_session = [ARSession new];
+		ar_delegate = [ARKitSessionDelegate new];
+		[ar_delegate setInterface:this];
+		ar_session.delegate = ar_delegate;
+
+		// reset our transform
+		transform = Transform();
+
+		// make this our primary interface
+		arvr_server->set_primary_interface(this);
+
+		// yeah!
+		initialized = true;
+
+		// Start our session...
+		start_session();
+	}
+
+	return true;
+}
+
+void ARKitInterface::uninitialize() {
+	if (initialized) {
+		ARVRServer *arvr_server = ARVRServer::get_singleton();
+		if (arvr_server != NULL) {
+			// no longer our primary interface
+			arvr_server->clear_primary_interface_if(this);
+		}
+
+		remove_all_anchors();
+
+		[ar_session release];
+		[ar_delegate release];
+		ar_session = NULL;
+		ar_delegate = NULL;
+		initialized = false;
+		session_was_started = false;
+	}
+}
+
+Size2 ARKitInterface::get_render_targetsize() {
+	_THREAD_SAFE_METHOD_
+
+	Size2 target_size = OS::get_singleton()->get_window_size();
```

Comment:
Is the window always the same size as the `Viewport`? Just thinking of edge cases here.

Reply:
Yes should be